NuRadioReco/examples/cr_efficiency_analysis/1_threshold_estimate.py

```diff
@@ -0,0 +1,284 @@
+channelGenericNoiseAdder = NuRadioReco.modules.channelGenericNoiseAdder.channelGenericNoiseAdder()
+channelGenericNoiseAdder.begin()
+channelGalacticNoiseAdder = NuRadioReco.modules.channelGalacticNoiseAdder.channelGalacticNoiseAdder()
+channelGalacticNoiseAdder.begin(n_side=4, interpolation_frequencies=np.arange(10, 1100, galactic_noise_interpolation_frequencies_step) * units.MHz)
+hardwareResponseIncorporator = NuRadioReco.modules.RNO_G.hardwareResponseIncorporator.hardwareResponseIncorporator()
+channelBandPassFilter = NuRadioReco.modules.channelBandPassFilter.channelBandPassFilter()
+channelBandPassFilter.begin()
+
+if trigger_name == 'high_low':
+    triggerSimulator = NuRadioReco.modules.trigger.highLowThreshold.triggerSimulator()
+    triggerSimulator.begin()
+
+if trigger_name == 'envelope':
+    triggerSimulator = NuRadioReco.modules.trigger.envelopeTrigger.triggerSimulator()
+    triggerSimulator.begin()
+
+t = time.time()  # absolute time of system
+sampling_rate = station.get_channel(channel_ids[0]).get_sampling_rate()
+dt = 1. / sampling_rate
+# print('samling rate', sampling_rate)
+# print('1/dt', dt)
+
+time = channel.get_times()
+channel_trace_start_time = time[0]
+channel_trace_final_time = time[len(time)-1]
+channel_trace_time_interval = channel_trace_final_time - channel_trace_start_time
+
+trigger_status = []
+triggered_trigger = []
+trigger_rate = []
+trigger_efficiency = []
+thresholds = []
+iterations = []
+channel_rms = []
+channel_sigma = []
+
+for n_thres in count():
+    threshold = threshold_start + (n_thres * threshold_step)
+    thresholds.append(threshold)
+    trigger_status_per_all_it = []
+
+    for n_it in range(n_iterations):
+        station = event.get_station(default_station)
+        eventTypeIdentifier.run(event, station, "forced", 'cosmic_ray')
+        for channel in station.iter_channels():
+            default_trace = np.zeros(1024)
+            channel.set_trace(trace=default_trace, sampling_rate=sampling_rate)
+
+        channelGenericNoiseAdder.run(event, station, det, amplitude=Vrms_thermal_noise, min_freq=T_noise_min_freq, max_freq=T_noise_max_freq, type='rayleigh')
+        channelGalacticNoiseAdder.run(event, station, det)
+        if hardware_response == True:
+            hardwareResponseIncorporator.run(event, station, det, sim_to_data=True)
+
+        for i_phase in range(10):
+            print('test iteration', (i_phase) + (n_it*10))
+            trigger_status_one_it = []
+            for channel in station.iter_channels():
+                # print('change phase in channel', channel.get_id())
+                freq_specs = channel.get_frequency_spectrum()
+                rand_phase = np.random.uniform(low=0, high= 2*np.pi, size=len(freq_specs))
+                freq_specs = np.abs(freq_specs) * np.exp(1j * rand_phase)
+                channel.set_frequency_spectrum(frequency_spectrum=freq_specs, sampling_rate=sampling_rate)
+                if trigger_name == 'high_low':
```

Comment:
Because the passband choice is inclueded in the envelope trigger module. For the highlow trigger, there is no such option. If the high-low trigger should be applied only to a certain passband, i need to add an extra trigger.

Reply:
Okay, fine with me. But maybe add a comment explaining your reasoning

Reply:
okay :)